### PR TITLE
Mirror summary for BrooklynEntityMirror

### PR DESCRIPTION
**Pull request by neykov**
Also extract the catalogItemId in a separate attribute for easier access.

### CODE REVIEW

File: software/base/src/main/java/brooklyn/entity/brooklynnode/BrooklynEntityMirror.java [hunk 1]
```diff
@@ -54,6 +56,9 @@
         "Frequency to poll for client sensors", Duration.FIVE_SECONDS);
     
     public static final AttributeSensor<String> MIRROR_STATUS = Sensors.newStringSensor("brooklyn.mirror.monitoring_status");
+    @SuppressWarnings("rawtypes")
+    public static final AttributeSensor<Map> MIRROR_SUMMARY = Sensors.newSensor(Map.class, "brooklyn.mirror.summary");
+    public static final AttributeSensor<String> MIRROR_CATALOG_ITEM_ID = Sensors.newStringSensor("brooklyn.mirror.catalog_item_id");
```
Comment (ahgittin) [inline software/base/src/main/java/brooklyn/entity/brooklynnode/BrooklynEntityMirror.java]: add description, noting that this is the catalog item id in the remote brooklyn of the mirrored entity, which could be different to the local catalog item id

File: software/base/src/main/java/brooklyn/entity/brooklynnode/BrooklynEntityMirrorImpl.java [hunk 2]
```diff
@@ -107,15 +107,25 @@ public Void apply(HttpToolResponse input) {
                 return null;
             }
         };
-
-        String sensorsUri = Urls.mergePaths(mirroredEntityUrl, "sensors/current-state");
+        
+        @SuppressWarnings("rawtypes")
+        Function<HttpToolResponse, Map> mirrorSummary = new Function<HttpToolResponse, Map>() {
+            @Override
+            public Map apply(HttpToolResponse input) {
```
Comment (ahgittin) [inline software/base/src/main/java/brooklyn/entity/brooklynnode/BrooklynEntityMirrorImpl.java]: better not to use an anonymous class?

File: software/base/src/main/java/brooklyn/entity/brooklynnode/BrooklynEntityMirror.java [hunk 3]
```diff
@@ -54,6 +56,9 @@
         "Frequency to poll for client sensors", Duration.FIVE_SECONDS);
     
     public static final AttributeSensor<String> MIRROR_STATUS = Sensors.newStringSensor("brooklyn.mirror.monitoring_status");
+    @SuppressWarnings("rawtypes")
+    public static final AttributeSensor<Map> MIRROR_SUMMARY = Sensors.newSensor(Map.class, "brooklyn.mirror.summary");
```
Comment (ahgittin) [inline software/base/src/main/java/brooklyn/entity/brooklynnode/BrooklynEntityMirror.java]: add description, to say that this is the json map returned by the entity rest endpoint (ie the `EntitySummary` REST model)

**Comment by ahgittin** [comment]
minor comments, then good to merge

**Comment by neykov** [comment]
Addressed comments, thanks @ahgittin

**Pull request merged**